include/swift/AST/Types.h

```diff
@@ -1804,12 +1804,10 @@ class TypeAliasType final
   }
 
   /// Get the innermost generic arguments, which correspond to the generic
-  /// arguments that are directly applied to the typealias declaration in
-  /// produced by \c getDecl().
-  ///
-  /// The result can be empty, if the declaration itself is non-generic but
-  /// the parent is generic.
-  SmallVector<Type, 2> getInnermostGenericArgs() const;
+  /// arguments that are directly applied to the typealias declaration
+  /// produced by \c getDecl(). Generic arguments applied to a parent
+  /// type are not included.
+  ArrayRef<Type> getInnermostGenericArgs() const;
```

Comment:
I am not a fan of `getInnermostGenericArgs` though. "Direct" generic arguments and innermost generic arguments are not always the same thing.